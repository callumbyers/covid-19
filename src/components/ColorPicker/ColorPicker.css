#CountryColorLabel, #CountryColorChoiceLabel {
    font-size: 16pt;
    color: #AD8175;
}

#ColorPickerHeader {
    font-size: 18pt;
    color: white;
    font-weight: 200;
    padding-top: 40px;
    margin-bottom: 40px;
}

.ColorPickerSelect {
    cursor: pointer;
    margin-top: 40px;
}

.SaveButton {
    margin-top: 40px !important;
    background-color: #679775 !important;
}

#ColorPickerContainer {
    width: 1000px;
    height: 600px;
    background-color: #3B3B3D;
    margin-top: 40px;
}

@media only screen and (max-width: 600px) {
    #ColorPickerContainer {
        width: unset;
    }
}