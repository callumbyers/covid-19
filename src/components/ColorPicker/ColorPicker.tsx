import React, { useContext, useState } from 'react';
import { SimpleSelect } from '../../elements';
import { AppContext } from '../../state/context';
import { InputLabel, Button } from '@material-ui/core';
import { colorOptions } from '../../consts';
import { updateAppData, updateCountryColor } from '../../state/actionCreators';
import './ColorPicker.css';

const colorPickerStyle = {
    width: 800,
    height: 600,
    backgroundColor: '#3B3B3D',
    marginTop: 40,
};

const ColorPicker = () => {
    const { state, dispatch } = useContext(AppContext);
    const [country, setCountry] = useState('');
    const [color, setColor] = useState('');

    const updateSelectedCountry = (value: string) => {
        setCountry(value);
    };

    const updateSelectedColor = (value: string) => {
        setColor(value);
    };

    const saveColor = () => {
        if (country && color) {
            const newCountry = state.appData.find((r) => r.name === country);
            if (newCountry) {
                newCountry.color = color;
                updateCountryColor(newCountry, dispatch);
            }
        }
    };

    return (
        <div id="ColorPickerContainer">
            <div id="ColorPickerHeader">Change Country Color</div>
            <div className="ColorDropdown">
                <InputLabel shrink id="CountryColorLabel">
                    Country
                </InputLabel>
                <SimpleSelect
                    labelId="CountryColorLabel"
                    className="ColorPickerSelect"
                    value={country}
                    onChange={updateSelectedCountry}
                    label={'Country'}
                    placeholder="Select Country"
                    options={state.countryOptions || []}
                />
            </div>
            <div className="ColorDropdown">
                <InputLabel shrink id="CountryColorChoiceLabel">
                    Color
                </InputLabel>
                <SimpleSelect
                    labelId="CountryColorChoiceLabel"
                    className="ColorPickerSelect"
                    value={color}
                    onChange={updateSelectedColor}
                    label={'Color'}
                    placeholder="Select Color"
                    options={colorOptions}
                />
            </div>
            <Button onClick={saveColor} className="SaveButton">
                Save
            </Button>
        </div>
    );
};

export default ColorPicker;
