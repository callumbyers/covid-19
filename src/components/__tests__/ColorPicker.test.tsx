import * as React from 'react';
import ColorPicker from '../ColorPicker/ColorPicker';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(<ColorPicker />).toJSON();
    expect(tree).toMatchSnapshot();
});
