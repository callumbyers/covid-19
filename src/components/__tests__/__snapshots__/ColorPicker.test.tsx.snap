// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="ColorPickerContainer"
>
  <div
    id="ColorPickerHeader"
  >
    Change Country Color
  </div>
  <div
    className="ColorDropdown"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-shrink"
      data-shrink={true}
      id="CountryColorLabel"
    >
      Country
    </label>
    <div>
      <div
        className="MuiFormControl-root makeStyles-formControl-3"
      >
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-customSelect-1 MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="SingleSelectLabel SingleSelectInput"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            id="SingleSelectInput"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            onAnimationStart={[Function]}
            placeholder="Select Country"
            type="hidden"
            value=""
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ColorDropdown"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-shrink"
      data-shrink={true}
      id="CountryColorChoiceLabel"
    >
      Color
    </label>
    <div>
      <div
        className="MuiFormControl-root makeStyles-formControl-3"
      >
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-customSelect-1 MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="SingleSelectLabel SingleSelectInput"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            id="SingleSelectInput"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            onAnimationStart={[Function]}
            placeholder="Select Color"
            type="hidden"
            value=""
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text SaveButton"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Save
    </span>
  </button>
</div>
`;
